backend/api/serializers/computetask.py

```diff
@@ -258,17 +285,44 @@ def create(self, validated_data):
         outputs = validated_data.pop("outputs")
 
         compute_task = super().create(validated_data)
+        input_kinds = {algo_input.identifier: algo_input.kind for algo_input in compute_task.algo.inputs.all()}
 
         for order, data_sample in enumerate(data_samples):
             TaskDataSamples.objects.create(compute_task=compute_task, data_sample=data_sample, order=order)
 
         for position, input in enumerate(inputs):
-            ComputeTaskInput.objects.create(
+            task_input = ComputeTaskInput.objects.create(
                 channel=compute_task.channel,
                 task=compute_task,
                 position=position,
                 **input,
             )
+            # task input asset could be known during task registration
+            # or could be resolved later if it does not exist yet
+            if task_input.asset_key:
+                ComputeTaskInputAsset.objects.create(
+                    channel=compute_task.channel,
+                    task_input=task_input,
+                    asset_key=task_input.asset_key,
+                    asset_kind=input_kinds[task_input.identifier],
+                )
+            elif task_input.parent_task_key:
+                task_output = ComputeTaskOutput.objects.get(
+                    task_id=task_input.parent_task_key,
+                    identifier=task_input.identifier,
+                )
+                try:
+                    task_output_asset = task_output.assets.filter(
+                        asset_kind=task_input.parent_task_output_identifier,
+                    )[position]
```

Comment:
It seems that we used to get the first asset (see [here](https://github.com/Substra/substra-backend/pull/458/files#diff-9edf1f25f3fa8d5bae29bf35657c7d05f03fb33b6985d7560153d319fc7fc0b2L354))